kyuubi-server/src/main/scala/org/apache/kyuubi/server/statestore/SessionStateStore.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.statestore
+
+import java.util.concurrent.TimeUnit
+
+import scala.collection.JavaConverters._
+
+import org.apache.kyuubi.client.api.v1.dto.{Batch, BatchRequest}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.engine.ApplicationOperation._
+import org.apache.kyuubi.server.KyuubiRestFrontendService
+import org.apache.kyuubi.server.statestore.api.{Batch => BatchState, BatchRequest => BatchRequestState}
+import org.apache.kyuubi.service.AbstractService
+import org.apache.kyuubi.util.ThreadUtils
+
+class SessionStateStore extends AbstractService("SessionStateStore") {
+  private var _stateStore: StateStore = _
+
+  private lazy val restInstance = KyuubiRestFrontendService.getConnectionUrl
+  private val stateStoreCleaner =
+    ThreadUtils.newDaemonSingleThreadScheduledExecutor("session-state-store-cleaner")
+
+  override def initialize(conf: KyuubiConf): Unit = {
+    this.conf = conf
+    _stateStore = StateStoreProvider.createStateStore(conf)
+    super.initialize(conf)
+  }
+
+  override def start(): Unit = {
+    super.start()
+    startStateStoreCleaner()
+  }
+
+  override def stop(): Unit = {
+    super.stop()
+    ThreadUtils.shutdown(stateStoreCleaner)
+  }
+
+  def getStateStore: StateStore = _stateStore
+
+  def createBatch(
+      batchId: String,
+      batchOwner: String,
+      conf: Map[String, String],
+      batchRequest: BatchRequest): Unit = {
+    val batchState = BatchState(
+      batchId,
+      batchRequest.getBatchType,
+      batchOwner,
+      conf,
+      restInstance,
+      "PENDING",
```

Comment:
waiting for #2699

Reply:
and shall update the state to `RUNNING` if applicationState got at the first time.